include/dr/mhp/algorithms/inclusive_exclusive_scan_impl.hpp

```diff
@@ -55,22 +69,37 @@ auto inclusive_scan_impl_(R &&r, O &&d_first, BinaryOp &&binary_op,
     if (dr::ranges::rank(global_in) == rank) {
       auto local_in = dr::ranges::__detail::local(global_in);
       auto local_out = dr::ranges::__detail::local(global_out);
+      if (is_exclusive && seg_index != 0) {
+        auto input_seg = input[seg_index - 1];
+        auto input_seg_size = rng::size(input_seg);
+        auto input_value = input_seg[input_seg_size - 1];
+        init = {input_value};
+      }
       if (use_sycl) {
```

Comment:
use_sycl is true only if SYCL_LANGUAGE_VERSION is defined (see global.hpp:37). The condition, together with #ifdef seems redundant.